### ISSUE TITLE
Fix zombie processes not closing in development

### ISSUE THREAD

**Issue by tjcouch-sil** [paranext/paranext-core#231]
It seems sometimes some node processes don't get closed in development. I theorize it may be one of the processes created in [this area](https://github.com/paranext/paranext-core/blob/fe9ec11c187af13a3c430e1e93e29e09bab4d049/.erb/configs/webpack.config.renderer.dev.ts#L215) as it seems the process itself isn't being listened to to close the other processes, and the spawned npm process at this line doesn't get saved off to get closed either. This is an issue with [ERB](https://github.com/electron-react-boilerplate/electron-react-boilerplate) itself, I believe, so it may be worth adding an issue or PR to them as well.

Sometimes esbuild processes don't get closed either, but I don't know if we are going to figure that out... Seems like a Vite issue.

Fixing this may help with #186

**Comment by irahopkinson** [comment]
The most common issue I see is it just stops after vite2 (probably due to the EPERM error). I'm running in a seperate terminal Windows Terminal) and my console looks typically like this:
![image](https://github.com/paranext/paranext-core/assets/5582153/8d0560ba-0c0e-4c43-8dd3-29cd4e35369d)
And my task manager looks like this:
![image](https://github.com/paranext/paranext-core/assets/5582153/78bb3b46-5e2d-41db-9841-55c9112525c6)
I've only recently been monitoring the task manager and in this state I just kill both `esbuild.exe` and all the `Node.js JavaScript Runtime` tasks and I can try again.

**Comment by tjcouch-sil** [comment]
> The most common issue I see is it just stops after vite2 (probably due to the EPERM error). I'm running in a seperate terminal Windows Terminal) and my console looks typically like this: ![image](https://user-images.githubusercontent.com/5582153/245993542-8d0560ba-0c0e-4c43-8dd3-29cd4e35369d.png) And my task manager looks like this: ![image](https://user-images.githubusercontent.com/5582153/245993747-78bb3b46-5e2d-41db-9841-55c9112525c6.png) I've only recently been monitoring the task manager and in this state I just kill both `esbuild.exe` and all the `Node.js JavaScript Runtime` tasks and I can try again.

@irahopkinson any news on trying again after closing those rogue processes that may have been there since before the changes? I haven't seemed to get the esbuild or node processes to zombie yet since my last changes, but I haven't tried extensively by any means.

**Comment by tjcouch-sil** [comment]
Ooooh it seems Task Manager doesn't update to show the new zombie processes. You have to close and open it. Bummer. After editing things in bundled extensions and the extension template back to back many times, I finally got an EPERM. Here are the command line args for all the rogue node processes. There were also two rogue esbuild processes.

![image](https://github.com/paranext/paranext-core/assets/104016682/d83f6950-e8eb-4324-a4f3-6c9b5a2823c9)

**Comment by tjcouch-sil** [comment]
Looks like `concurrently` has a command-line argument `--kill-others-on-fail` that may do exactly what we want!

**Comment by tjcouch-sil** [comment]
Looks like `--kill-others-on-fail` doesn't solve all our problems unfortunately. Some other ideas:

- Try/catch closing processes in webpack renderer dev? Might have to do with stuff not closing
  - Tried this, and it doesn't work. But I added some debug logs. Looks like maybe not all the processes get closed here - I wonder if they are already gone or are not being closed in the right place or something. Getting ENOSYS on trying to kill the processes, and different processes throw the error when reproducing different times:
```
Main is closing!
webpack.config.renderer.dev.ts:239
Preload Builder threw an error during closing of Main! Error: kill ENOSYS
webpack.config.renderer.dev.ts:228
Extension Builder threw an error during closing of Main! Error: kill ENOSYS
```
- Are we killing extension host and dotnet provider (launched from main) on crash? `extension-host.service.ts` and `dotnet-data-provider.service.ts` launch them from `main.ts`
- Does killing extension host kill all its processes? `extension-host.service.ts`
- At some point in the past, Tim and I worked on similar problems in #155, and we discovered that running `npm run ...` to launch the extension host watcher `nodemon` was failing to close everything properly. We changed to [executing the script we were trying to run manually with `node ...`](https://github.com/paranext/paranext-core/pull/155/files#diff-d89b4285a98ce297f1ad6a2ef4ef247bf90cb3e1ff1a9560c81ba262be65923bL57) instead. Though this may have had at least something to do with the fact that it was running a watcher script, but I'm really not sure. Needs more investigation. Anyway, that could be a culprit in `webpack.config.renderer.dev.ts` where it's launching many processes and then trying to close them later.
- Maybe it's worth running `concurrently` on the many processes launched in `webpack.config.renderer.dev.ts` instead of launching them inside the webpack config?

Notes:
- If you see `esbuild` running in Task Manager, that means Vite didn't stop properly in some way
